cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/file/FileAdmin.java

```diff
@@ -32,9 +36,13 @@
 
   private final String basePath;
 
-  public FileAdmin(LocationFactory locationFactory, DatasetSpecification spec) {
+  public FileAdmin(CConfiguration cConf, LocationFactory locationFactory, DatasetSpecification spec) {
+    TableId tableId = TableId.from(spec.getName());
```

Comment:
Due to the current drawback in the design mentioned in CDAP-1593, the ``DatasetAdmin`` and ``Dataset`` classes use the ``TableId`` class to find out the namespace that the table has to go in, and its table name (in HBase). It has the capability to handle backwards compatibility in table names (from 2.6, when there were no namespaces).

This will go away when ``FileAdmin`` can get an ``Id.DatasetInstance`` as another parameter, which is the work that I'm starting now.